### PR TITLE
Course/Certificate: Fix Mantis issue 29191

### PR DESCRIPTION
**Pull request by mjansenDatabay**
If approved, we'll have to add a `ilCtrl` reload to read the `service.xml` changes.

Mantis Issue: https://mantis.ilias.de/view.php?id=29191

Original PR: #3034

### CODE REVIEW

**Comment by klees** [comment]
Hi @smeyer-ilias,

we stumbled upon this PR in our TB meeting. Any updates on this?

Best regards!

**Comment by kergomard** [comment]
Hi @smeyer-ilias 
As Technical Board, we regularly check for pull requests that have been open for a long time.
Any Updates on this?

Best regards!

**Comment by Amstutz** [comment]
Hi @smeyer-ilias
As Technical Board, we regularly check for pull requests that have been open for a long time.
Any Updates on this?

Best regards!

**Comment by Amstutz** [comment]
Hi @smeyer-ilias
As Technical Board, we regularly check for pull requests that have been open for a long time.
Any Updates on this?

Best regards!

**Comment by Amstutz** [comment]
Hi @smeyer-ilias
As Technical Board, we regularly check for pull requests that have been open for a long time.
Any Updates on this?

Best regards!

**Comment by Amstutz** [comment]
Hi @smeyer-ilias
As Technical Board, we regularly check for pull requests that have been open for a long time.
Any Updates on this?

Best regards!

**Comment by Amstutz** [comment]
Hi @smeyer-ilias
As Technical Board, we regularly check for pull requests that have been open for a long time.
Any Updates on this?

Best regards!

**Comment by smeyer-ilias** [comment]
Sorry, for the delay.
I will check and test this PR after my vacation.

File: Services/Tracking/classes/status/class.ilLPStatusCollection.php [hunk 1]
```diff
@@ -323,6 +329,44 @@ public static function determineGroupingStatus(
...
+    /**
+     * @param int $objId
+     * @param int $usrId
+     * @return bool
+     */
+    protected function isMember(int $objId, int $usrId) : bool
+    {
+        global $DIC;
+
+        $ilObjDataCache = $DIC['ilObjDataCache'];
```
Comment (smeyer-ilias) [inline Services/Tracking/classes/status/class.ilLPStatusCollection.php]: Please use $this->ilObjDataCache

File: Services/Tracking/classes/status/class.ilLPStatusCollection.php [hunk 2]
```diff
@@ -323,6 +329,44 @@ public static function determineGroupingStatus(
...
+        $tree = $DIC['tree'];
```
Comment (smeyer-ilias) [inline Services/Tracking/classes/status/class.ilLPStatusCollection.php]: Please introduce $tree in the constructor of ilLPCollection.

File: Services/Tracking/classes/status/class.ilLPStatusCollection.php [hunk 3]
```diff
@@ -323,6 +329,44 @@ public static function determineGroupingStatus(
...
+                return $participants->isMember();
+
+            case 'grp':
+                $participants = ilGroupParticipants::_getInstanceByObjId($objId);
+                return $participants->isMember($usrId);
+
+            case 'fold':
+                $folderRefIds = ilObject::_getAllReferences($objId);
+                $folderRefId = current($folderRefIds);
+                if ($crsRefId = $tree->checkForParentType($folderRefId, 'crs')) {
```
Comment (smeyer-ilias) [inline Services/Tracking/classes/status/class.ilLPStatusCollection.php]: Please implement an additional check for group membership here: checkForParentType($foldeRefId, 'grp');

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]

**Pull request merged**